lib/board.rb

```diff
@@ -0,0 +1,66 @@
+class Board
+  attr_accessor :cells, :display, :position, :turn_count, :update
+  def initialize
+    @cells = Array.new(9, " ")      #has an attribute to store the cells of the board
+  end
+
+  def reset!                      #can reset the state of the cells in the board
+    @cells = Array.new(9, " ")    #sets the cells of the board to a 9 element array of " "
+  end
+
+  def display   #prints the board
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(board_cell)                #takes in user input and returns the value of the board cell
+      board_cell = board_cell.to_i-1
+      @cells[board_cell]
+  end
+
+  def full?    #returns true for a full board, returns false for an in-progress game
```

Comment:
Use the most expressive and efficient iterator for a problem. `Board#full?` is asking if all elements are not empty strings. You can use logic that is symmetrical to those semantics with the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`

This is also more efficient in that the loop will stop after the iteration for which a cell is not empty.  Also, try not to explicitly return a boolean from an if statement if not needed.